pkgs/development/python-modules/cymem/default.nix

```diff
@@ -2,32 +2,30 @@
 , buildPythonPackage
 , fetchFromGitHub
 , cython
-, unittestCheckHook
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
   pname = "cymem";
-  version = "2.0.3";
+  version = "2.0.6";
 
   src = fetchFromGitHub {
     owner = "explosion";
     repo = "cymem";
     rev = "v${version}";
-    sha256 = "0cqz6whq4zginxjnh4cfqlsh535p4qz295ymvjchp71fv8mz11f6";
+    sha256 = "sha256-o+44v6wvE9HxeQaDDQ0+gi7z1V7jtkZvWglY8UyVHLg=";
   };
 
   propagatedBuildInputs = [
     cython
   ];
 
-  prePatch = ''
-    substituteInPlace setup.py \
-      --replace "wheel>=0.32.0,<0.33.0" "wheel>=0.31.0"
-  '';
+  # ModuleNotFoundError: No module named 'cymem.cymem'
```

Comment:
We should probably improve our check hooks by moving out of the source directory. Following is a work-around, but it works because I am using the absolute path to the source here.
```
diff --git a/pkgs/development/python-modules/cymem/default.nix b/pkgs/development/python-modules/cymem/default.nix
index 88e25720109..83f0ae05595 100644
--- a/pkgs/development/python-modules/cymem/default.nix
+++ b/pkgs/development/python-modules/cymem/default.nix
@@ -20,12 +20,12 @@ buildPythonPackage rec {
     cython
   ];
 
-  # ModuleNotFoundError: No module named 'cymem.cymem'
-  doCheck = false;
-
+  preCheck = ''
+    cd $out
+  '';
   checkInputs = [ pytestCheckHook ];
 
-  pytestFlagsArray = [ "cymem" ];
+  pytestFlagsArray = [ "/build/source" ];
 
   meta = with lib; {
     description = "Cython memory pool for RAII-style memory management";

```

Reply:
I'm not sure if we can assume `/build`.  At least without sandboxing it won't work, I think.

Reply:
@FRidh Why does this fix the issue? Upstream does what we do (execute pytest from the source directory) [in CI](https://github.com/explosion/cymem/blob/1ec862e8f8369b17e114fb1cb026b087f8cf7f93/azure-pipelines.yml#L92), and it passes just fine.

@vcunat We can't assume it, no, for exactly that reason.

Reply:
Absolute path to the root is in some variable defined by nix, for sure, but that might be a bit more complicated to get expanded correctly.

Reply:
I think that can be worked around pretty easily (not to say it'll be clean, but it'll work), but I'd like to understand why we have to deviate from upstream (and other packages that use pytest) before implementing it.

Reply:
So if I cd into `cymem` in `preCheck`, everything works fine (without the `cymem` argument to pytest).

I just don't understand why this doesn't happen upstream... unless it's because they [install the package](https://github.com/explosion/cymem/blob/1ec862e8f8369b17e114fb1cb026b087f8cf7f93/azure-pipelines.yml#L82-L88) before running the tests -- which I guess is only required because this is a Cython module (otherwise `pytest` would fixup `sys.path` so the sources could be imported).

So if this is relying on our adding of the installed `cymem` to `PYTHONPATH`... why is moving into that directory required? Clearly the tests are discovered just fine, and the package is installed whether we run the `cd` or not... so upstream's execution method should work fine, considering we install it just like they do.

*why is this so confusing*

Reply:
> @FRidh Why does this fix the issue? Upstream does what we do (execute pytest from the source directory) [in CI](https://github.com/explosion/cymem/blob/1ec862e8f8369b17e114fb1cb026b087f8cf7f93/azure-pipelines.yml#L92), and it passes just fine, so I'm confused as to why we need to do that. Note that I've tried with and without the `cymem` argument, and get the same result.

That's because they build in place. Bad practice.

> python setup.py build_ext --inplace

We need to move out of the directory because Python adds the current working directory to `sys.path`.